DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.WebAPI/Queries/GitHubEventsQuery.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using GitHubExtension.Activity.External.WebAPI.Extensions;
+using GitHubExtension.Activity.External.WebAPI.Models;
+using GitHubExtension.Infrastructure.Extensions;
+using Newtonsoft.Json;
+
+namespace GitHubExtension.Activity.External.WebAPI.Queries
+{
+    class GitHubEventsQuery : IGitHubEventsQuery
+    {
+        private const string EventsUrl = "https://api.github.com/repos/{0}/events?access_token={1}&page={2}";
+        private readonly HttpClient _httpClient;
+
+        public GitHubEventsQuery()
+        {
+            _httpClient = new HttpClient();
+        }
+
+        public async Task<EventsPaginationModel> GetGitHubEventsAsync(string fullRepositoryName, string token, int numberOfPage)
+        {
+            string requestUri = string.Format(EventsUrl, fullRepositoryName, token, numberOfPage);
+            using (var message = new HttpRequestMessage(HttpMethod.Get, requestUri).AddHeadersForGitHub())
+            using (HttpResponseMessage response = await _httpClient.SendAsync(message))
+            {
+                int? numberOfPages = null;
```

Comment:
Please move to line 34. All local variables should be close to place where it used.